tests/index.test.js

```diff
@@ -92,3 +92,29 @@ test('pathing near close nodes in a different group', (t) => {
 
   t.end();
 });
+
+test('overlapping groups', (t) => {
```

Comment:
You're referring to vertically stacked groups, not intersecting groups, right? (the latter should not be allowed). I'd rename this test just to be clear there.